mlflow/server/js/src/components/MetricView.js

```diff
@@ -101,11 +116,16 @@ const mapStateToProps = (state, ownProps) => {
       metrics[i][runUuid] = entries[i].value;
     }
   });
+  const runNames = runUuids.map((runUuid) => {
```

Comment:
Creating objects in `mapStateToProps` makes it so you can never use PureComponent to optimize rendering. Every time redux store updates, this will create a new `runNames` object (and `metrics` object), which will cause shallowEquals of props to be false, triggering a render. Going forward you might want to calculate derived data in a selector, so it can be memoized with [reselect](https://www.npmjs.com/package/reselect).